RecoMuon/GlobalTrackingTools/plugins/GlobalTrackQualityProducer.cc

```diff
@@ -311,5 +311,77 @@ GlobalTrackQualityProducer::trackProbability(Trajectory& track) const {
+    descGlbMuonRefitter.add<edm::InputTag>("GEMRecHitLabel" , edm::InputTag("gemRecHits"));
+    descGlbMuonRefitter.add<edm::InputTag>("ME0RecHitLabel" , edm::InputTag("me0Segments"));
+    descGlbMuonRefitter.add<edm::InputTag>("RPCRecSegmentLabel" , edm::InputTag("rpcRecHits"));
+
+    descGlbMuonRefitter.add<int>("MuonHitsOption", 1);
+    descGlbMuonRefitter.add<double>("PtCut", 1.0);
+    descGlbMuonRefitter.add<double>("Chi2ProbabilityCut", 30.0);
+    descGlbMuonRefitter.add<double>("Chi2CutCSC", 1.0);
+    descGlbMuonRefitter.add<double>("Chi2CutDT", 30.0);
+    descGlbMuonRefitter.add<double>("Chi2CutGEM", 1.0);
+    descGlbMuonRefitter.add<double>("Chi2CutME0", 1.0);
+    descGlbMuonRefitter.add<double>("Chi2CutRPC", 1.0);
+    descGlbMuonRefitter.add<int>("HitThreshold", 1);
+
+    descGlbMuonRefitter.add<std::string>("Fitter", "KFFitterForRefitInsideOut");
+    descGlbMuonRefitter.add<std::string>("Smoother", "KFSmootherForRefitInsideOut");
+    descGlbMuonRefitter.add<std::string>("Propagator", "SmartPropagatorAnyRK");
+    descGlbMuonRefitter.add<std::string>("TrackerRecHitBuilder", "WithAngleAndTemplate");
+    descGlbMuonRefitter.add<std::string>("MuonRecHitBuilder", "MuonRecHitBuilder");
+    descGlbMuonRefitter.add<bool>("DoPredictionsOnly", false);
+    descGlbMuonRefitter.add<std::string>("RefitDirection", "insideOut");
+    descGlbMuonRefitter.add<bool>("PropDirForCosmics", false);
+    descGlbMuonRefitter.add<bool>("RefitRPCHits", true);
+
+    descGlbMuonRefitter.add<std::vector<int>>("DYTthrs",{10, 10});
+    descGlbMuonRefitter.add<int>("DYTselector",1);
+    descGlbMuonRefitter.add<bool>("DYTupdator", false);
+    descGlbMuonRefitter.add<bool>("DYTuseAPE", false );
+    descGlbMuonRefitter.add<bool>("DYTuseThrsParametrization", false);
+    {
+      edm::ParameterSetDescription descDYTthrs;
+      descDYTthrs.add<std::vector<double>>("eta0p8", {1,-0.919853, 0.990742});
+      descDYTthrs.add<std::vector<double>>("eta1p2", {1,-0.897354, 0.987738});
+      descDYTthrs.add<std::vector<double>>("eta2p0", {1,-0.986855, 0.998516});
+      descDYTthrs.add<std::vector<double>>("eta2p2", {1,-0.940342, 0.992955});
+      descDYTthrs.add<std::vector<double>>("eta2p4", {1,-0.947633, 0.993762});
+      descGlbMuonRefitter.add<edm::ParameterSetDescription>("DYTthrsParameters", descDYTthrs);
+    }
+
+    descGlbMuonRefitter.add<int>("SkipStation", -1);
+    descGlbMuonRefitter.add<int>("TrackerSkipSystem",-1);
+    descGlbMuonRefitter.add<int>("TrackerSkipSection", -1);
+    descGlbMuonRefitter.add<bool>("RefitFlag", true );
+
+    //descRefitter.add<edm::ParameterSetDescription>("GlobalMuonRefitter", descGlbMuonRefitter);
+    desc.add<edm::ParameterSetDescription>("RefitterParameters", descGlbMuonRefitter);
+  }
+  //desc.add<edm::ParameterSetDescription>("RefitterParameters", descRefitter);
+  desc.add<double>("nSigma", 3.0);
+  desc.add<double>("MaxChi2", 100000.0);
+
+  descriptions.add("GlobalTrackQualityProducer", desc);
```

Comment:
the default name should start with a lower case ```globalTrackQualityProducer```.

Perhaps more importantly: we should start using the file generated with the fillDescriptions to replace the content of RecoMuon/GlobalTrackingTools/python/GlobalTrackQuality_cfi.py